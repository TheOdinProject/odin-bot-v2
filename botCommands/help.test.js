const commands = require('./help');
const { generateMentions } = require('./mockData');

describe('?help', () => {
  it.each([
    ['?help'],
    [' ?help'],
    ['?help @odin-bot'],
    ['?help me please'],
    ['?help @user'],
    [' ?help @user'],
  ])("'%s' triggers the callback", (string) => {
    expect(commands.help.regex.test(string)).toBeTruthy();
  });

  it.each([
    ['?hmadpofdsnja'],
    ['?halp'],
    ['help'],
    ['?hlpe'],
    ['?he lp'],
    ['? help'],
    ['@odin-bot ? help'],
    ['?he&l^p'],
    ['?h*el&p^'],
    ['?^help'],
  ])("'%s' does not trigger the callback", (string) => {
    expect(commands.help.regex.test(string)).toBeFalsy();
  });

  it.each([
    ['send ?help'],
    ['need some ?help ?'],
    ['hey @odin-bot, teach me ?help'],
    ['?@odin-bot ^ ?me ?help ?test$*'],
  ])("'%s' - the command can be anywhere in the string", (string) => {
    expect(commands.help.regex.test(string)).toBeTruthy();
  });

  it.each([
    ['?helpsgrealkdmsfalnd'],
    ['someone?help'],
    ['?help?xx'],
    ['?help*'],
    ['?help...'],
    ['^_?help...'],
  ])(
    "'%s' - command should be its own word?group - no leading or trailing characters",
    (string) => {
      expect(commands.help.regex.test(string)).toBeFalsy();
    },
  );
});

describe('?help snapshot', () => {
  it('should return the correct output', () => {
    expect(commands.help.cb({})).toMatchSnapshot();
  });
});

describe('?code', () => {
  it.each([
    ['?code'],
    [' ?code'],
    ['?code @odin-bot'],
    [' ?code @odin-bot'],
    ['?code fmdkslafmksa'],
  ])("'%s' triggers the callback", (string) => {
    expect(commands.code.regex.test(string)).toBeTruthy();
  });

  it.each([
    ['?cfdsdnjae'],
    ['?cood'],
    ['code'],
    ['?cdoe'],
    ['?co de'],
    ['? code'],
    ['@odin-bot ? code'],
    ['?co&d^e'],
    ['?c*od&e^'],
    ['?^code'],
  ])("'%s' does not trigger the callback", (string) => {
    expect(commands.code.regex.test(string)).toBeFalsy();
  });

  it.each([
    ['how to share ?code'],
    ['need some ?code ?'],
    ['hey @odin-bot, teach me to ?code'],
    ['?@odin-bot ^ ?me ?code ?test$*'],
  ])("'%s' - the command can be anywhere in the string", (string) => {
    expect(commands.code.regex.test(string)).toBeTruthy();
  });

  it.each([
    ['?codesgrealkdmsfalnd'],
    ['someone?code'],
    ['@odin-bot?code'],
    ['?code?xx'],
    ['?code*'],
    ['?code...'],
    ['^_?code...'],
  ])(
    "'%s' - command should be its own word/group - no leading or trailing characters",
    (string) => {
      expect(commands.code.regex.test(string)).toBeFalsy();
    },
  );
});

describe('?code snapshot', () => {
  it('returns correct output', () => {
    expect(commands.code.cb(generateMentions(0))).toMatchSnapshot();
    expect(commands.code.cb(generateMentions(1))).toMatchSnapshot();
    expect(commands.code.cb(generateMentions(2))).toMatchSnapshot();
    expect(commands.code.cb(generateMentions(3))).toMatchSnapshot();
  });
});
