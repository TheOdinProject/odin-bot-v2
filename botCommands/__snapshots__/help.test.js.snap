// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`!code snapshot returns correct output 1`] = `
Object {
  "embeds": Array [
    Object {
      "color": 39423,
      "description": "Hey, here's some helpful tips on sharing your code with others!",
      "fields": Array [
        Object {
          "name": "Sharing Code on Discord",
          "value": "To write multiple lines of code with language syntax highlighting, use three backticks (<https://i.stack.imgur.com/ETTnT.jpg>), followed by the language.

\\\\\`\\\\\`\\\\\`js
[Put your JavaScript Code here!]
\\\\\`\\\\\`\\\\\`

For \`inline code\` use one backtick (no syntax highlighting):

\\\\\`Code here!\\\\\`
      ",
        },
        Object {
          "inline": true,
          "name": "Link a Code Sandbox to share Webpack/React examples",
          "value": "https://codesandbox.io/",
        },
        Object {
          "inline": true,
          "name": "Link a Repl.it to share Javascript/Ruby examples",
          "value": "https://replit.com/",
        },
        Object {
          "inline": true,
          "name": "Link a Codepen to share basic HTML/CSS/Javascript examples",
          "value": "https://codepen.io/",
        },
      ],
      "title": "HOW TO EMBED CODE SNIPPETS",
    },
  ],
}
`;

exports[`!code snapshot returns correct output 2`] = `
Object {
  "embeds": Array [
    Object {
      "color": 39423,
      "description": "Hey, <@1>, here's some helpful tips on sharing your code with others!",
      "fields": Array [
        Object {
          "name": "Sharing Code on Discord",
          "value": "To write multiple lines of code with language syntax highlighting, use three backticks (<https://i.stack.imgur.com/ETTnT.jpg>), followed by the language.

\\\\\`\\\\\`\\\\\`js
[Put your JavaScript Code here!]
\\\\\`\\\\\`\\\\\`

For \`inline code\` use one backtick (no syntax highlighting):

\\\\\`Code here!\\\\\`
      ",
        },
        Object {
          "inline": true,
          "name": "Link a Code Sandbox to share Webpack/React examples",
          "value": "https://codesandbox.io/",
        },
        Object {
          "inline": true,
          "name": "Link a Repl.it to share Javascript/Ruby examples",
          "value": "https://replit.com/",
        },
        Object {
          "inline": true,
          "name": "Link a Codepen to share basic HTML/CSS/Javascript examples",
          "value": "https://codepen.io/",
        },
      ],
      "title": "HOW TO EMBED CODE SNIPPETS",
    },
  ],
}
`;

exports[`!code snapshot returns correct output 3`] = `
Object {
  "embeds": Array [
    Object {
      "color": 39423,
      "description": "Hey, <@1> <@2>, here's some helpful tips on sharing your code with others!",
      "fields": Array [
        Object {
          "name": "Sharing Code on Discord",
          "value": "To write multiple lines of code with language syntax highlighting, use three backticks (<https://i.stack.imgur.com/ETTnT.jpg>), followed by the language.

\\\\\`\\\\\`\\\\\`js
[Put your JavaScript Code here!]
\\\\\`\\\\\`\\\\\`

For \`inline code\` use one backtick (no syntax highlighting):

\\\\\`Code here!\\\\\`
      ",
        },
        Object {
          "inline": true,
          "name": "Link a Code Sandbox to share Webpack/React examples",
          "value": "https://codesandbox.io/",
        },
        Object {
          "inline": true,
          "name": "Link a Repl.it to share Javascript/Ruby examples",
          "value": "https://replit.com/",
        },
        Object {
          "inline": true,
          "name": "Link a Codepen to share basic HTML/CSS/Javascript examples",
          "value": "https://codepen.io/",
        },
      ],
      "title": "HOW TO EMBED CODE SNIPPETS",
    },
  ],
}
`;

exports[`!code snapshot returns correct output 4`] = `
Object {
  "embeds": Array [
    Object {
      "color": 39423,
      "description": "Hey, <@1> <@2> <@3>, here's some helpful tips on sharing your code with others!",
      "fields": Array [
        Object {
          "name": "Sharing Code on Discord",
          "value": "To write multiple lines of code with language syntax highlighting, use three backticks (<https://i.stack.imgur.com/ETTnT.jpg>), followed by the language.

\\\\\`\\\\\`\\\\\`js
[Put your JavaScript Code here!]
\\\\\`\\\\\`\\\\\`

For \`inline code\` use one backtick (no syntax highlighting):

\\\\\`Code here!\\\\\`
      ",
        },
        Object {
          "inline": true,
          "name": "Link a Code Sandbox to share Webpack/React examples",
          "value": "https://codesandbox.io/",
        },
        Object {
          "inline": true,
          "name": "Link a Repl.it to share Javascript/Ruby examples",
          "value": "https://replit.com/",
        },
        Object {
          "inline": true,
          "name": "Link a Codepen to share basic HTML/CSS/Javascript examples",
          "value": "https://codepen.io/",
        },
      ],
      "title": "HOW TO EMBED CODE SNIPPETS",
    },
  ],
}
`;

exports[`!help snapshot should return the correct output 1`] = `
Object {
  "embeds": Array [
    Object {
      "color": 13407555,
      "description": "**By posting in this chatroom you agree to our [code of conduct](https://github.com/TheOdinProject/theodinproject/blob/master/doc/code_of_conduct.md).**",
      "fields": Array [
        Object {
          "inline": true,
          "name": "Add Points",
          "value": "Give points to someone who has been helpful by mentioning their name 
 and adding ++ : 
 \`@username ++\` or by 
 giving them a star : \`@username :star:\`.",
        },
        Object {
          "inline": true,
          "name": "Leaderboard",
          "value": "View the points leaderboard with \`!leaderboard\`. 

 Modify it with \`n=\` and \`start=\` i.e. \`!leaderboard n=25 start=30\`.",
        },
        Object {
          "inline": true,
          "name": "Contribute to the bot",
          "value": "<https://github.com/TheOdinProject/odin-bot-v2>",
        },
      ],
      "title": "Help",
    },
  ],
}
`;
